src/main/java/net/sf/mzmine/modules/visualization/spectra/simplespectra/spectraidentification/spectraldatabase/SpectralMatchTask.java

```diff
@@ -0,0 +1,242 @@
+  private int listsize;
+
+  public SpectralMatchTask(ParameterSet parameters, int startEntry, List<SpectralDBEntry> list,
+      SpectraPlot spectraPlot, Scan currentScan, SpectraIdentificationResultsWindow resultWindow) {
+    this.startEntry = startEntry;
+    this.list = list;
+    this.currentScan = currentScan;
+    this.spectraPlot = spectraPlot;
+    this.resultWindow = resultWindow;
+
+    listsize = list.size();
+    dataBaseFile = parameters
+        .getParameter(SpectraIdentificationSpectralDatabaseParameters.dataBaseFile).getValue();
+    massListName = parameters.getParameter(SpectraIdentificationSpectralDatabaseParameters.massList)
+        .getValue();
+    mzTolerance = parameters
+        .getParameter(SpectraIdentificationSpectralDatabaseParameters.mzTolerance).getValue();
+    noiseLevel = parameters.getParameter(SpectraIdentificationSpectralDatabaseParameters.noiseLevel)
+        .getValue();
+
+    minMatch = parameters.getParameter(SpectraIdentificationSpectralDatabaseParameters.minMatch)
+        .getValue();
+    minSimilarity = parameters
+        .getParameter(SpectraIdentificationSpectralDatabaseParameters.minCosine).getValue();
+  }
+
+  /**
+   * @see net.sf.mzmine.taskcontrol.Task#getFinishedPercentage()
+   */
+  @Override
+  public double getFinishedPercentage() {
+    if (totalSteps == 0)
+      return 0;
+    return ((double) finishedSteps) / totalSteps;
+  }
+
+  /**
+   * @see net.sf.mzmine.taskcontrol.Task#getTaskDescription()
+   */
+  @Override
+  public String getTaskDescription() {
+    return MessageFormat.format(
+        "(entry {2}-{3}) spectral database identification in {0} using database {1}", "spectrum",
+        dataBaseFile.getName(), startEntry, startEntry + listsize - 1);
+  }
+
+  /**
+   * @see java.lang.Runnable#run()
+   */
+  @Override
+  public void run() {
+    setStatus(TaskStatus.PROCESSING);
+    try {
+      totalSteps = list.size();
+      matches = new HashMap<SpectralDBEntry, Double>();
+      for (SpectralDBEntry ident : list) {
+        SpectraSimilarity sim = spectraDBMatch(currentScan, ident);
+        if (sim != null) {
+          count++;
+          matches.put(ident, sim.getCosine());
+        }
+        // check for max error (missing masslist)
+        if (errorCounter > MAX_ERROR) {
+          logger.log(Level.WARNING, "Spectral data base matching failed");
+          setStatus(TaskStatus.ERROR);
+          setErrorMessage("Spectral data base matching failed");
+          list = null;
+          return;
+        }
+        // next row
+        finishedSteps++;
+      }
+      addIdentities(matches);
+      logger.info("Added " + count + " spectral library matches");
+
+      // check if no match was found
+      if (count == 0) {
+        logger.log(Level.WARNING, "No data base matches found");
+        setErrorMessage("No data base matches found. Spectral data base matching failed");
+        list = null;
+        return;
+      }
+
+    } catch (Exception e) {
```

Comment:
Do not swallow exceptions - at least add ```e.printStackTrace()```.
Also, I think it is generally confusing to use long try/catch blocks. You should just take those parts of the code that may throw an exception, and surround those with try/catch.

Reply:
Maybe it is better to put this try-catch into the loop.
Otherwise the result of the task should be error

Reply:
Yeah, that was bad practice. I will add an exception and simplify the code.

Reply:
I will set it to error